### DISCUSSION TITLE
Development and Compilation time using CGAL

### DISCUSSION THREAD

**Discussion by petrasvestartas** [CGAL/cgal#6565] [Q&A] [answered]
Hi,

I am using CGAL daily for geometry modelling. I am loosing quite some time using C++ due to compilation time.
I do not know how much it is related to CGAL itself, but I am seeking your advice.

I tend to make a lot of small changes in Visual Studio, then I hit run to inspect a result and then come back to the code.
Usually it is just a few line changes of my own code. 

Is there any method to reduce the compilation time assuming I am not modifying any CGAL source files just a 3-4 headers of my own code during the day.

**Comment by lrineau** [comment, marked as answer]
You can use precompiled headers. The documentation for MSVC is here: https://docs.microsoft.com/en-us/cpp/build/creating-precompiled-header-files?view=msvc-170

The idea is to put all the CGAL headers you require in one header of your own, that would be precompiled.

**Reply by petrasvestartas** [reply]
Thank you, I never used them, hopefully this will work;)

**Reply by petrasvestartas** [reply]
From 27 seconds to 15 sec not bad two times the speed, thank you
